open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -91,7 +91,7 @@ def get_available_docker_host(self, req_count, hackathon):
         # The new-created VM must run 'cloudvm service by default(either cloud-init or python remote ssh)
         # todo the VM public/private IP will change after reboot, need sync the IP in db with azure in this case
         for docker_host in vms:
-            if self.hosted_docker.ping(docker_host):
+            if self.hosted_docker.ping(docker_host) & self.cloud_service_available.isAvailable(docker_host.public_ip):
```

Comment:
Do you have ever checked the output of `ListHostedService` in azure python sdk? Do we have to lock the object ourself?